mathesar/api/db/viewsets/records.py

```diff
@@ -103,11 +103,16 @@ def list(self, request, table_pk=None):
                 status_code=status.HTTP_400_BAD_REQUEST
             )
         except DataError as e:
-            if isinstance(e.orig, InvalidDatetimeFormat):
+            if isinstance(e.orig, (InvalidDatetimeFormat, DatetimeFieldOverflow)):
                 raise database_api_exceptions.InvalidDateFormatAPIException(
                     e,
                     status_code=status.HTTP_400_BAD_REQUEST,
                 )
+            else:
+                raise database_api_exceptions.RaiseExceptionAPIException(
```

Comment:
`MathesarAPIException` would be more appropriate here, check the following for reference.
https://github.com/mathesar-foundation/mathesar/blob/d375575199e81205ea9cd4a7a3b928398f43ec65/mathesar/api/serializers/records.py#L62-L63

Reply:
I have addressed all these changes that you mentioned in my latest commit.